### PR TITLE
gmail reader

### PR DESCRIPTION
**Pull request by bborn**
Add a document loader for gmail

### CODE REVIEW

File: loader_hub/gmail/base.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+"""Google Mail reader."""
+import os
+from googleapiclient.discovery import build
```
Comment (jerryjliu) [inline loader_hub/gmail/base.py:3]: would include as a lazy import

**Review by jerryjliu — CHANGES_REQUESTED** [review]
this looks great! 

i'd fill out the README.md and requirements.txt (seems like there's googleapiclient deps and bs4 deps

File: loader_hub/gmail/base.py (hunk 1 above)
Comment (bborn) [reply loader_hub/gmail/base.py]: Done

File: loader_hub/gmail/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+As a prerequisite, you will need to register with Google and generate a `credentials.json` file in the directory where you run this loader. See [here](https://developers.google.com/workspace/guides/create-credentials) for instructions.
+
+## Usage
+
+To use this loader, you simply need to pass in a search query string.
+
+```python
+from llama_index import download_loader
+
+GoogleDocsReader = download_loader('GmailReader')
```
Comment (jerryjliu) [inline loader_hub/gmail/README.md:14]: nit: GmailReader

File: loader_hub/gmail/README.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+## Examples
+
+This loader is designed to be used as a way to load data into [LlamaIndex](https://github.com/jerryjliu/gpt_index/tree/main/gpt_index) and/or subsequently used as a Tool in a [LangChain](https://github.com/hwchase17/langchain) Agent.
+
+### LlamaIndex
+
+```python
+from llama_index import GPTSimpleVectorIndex, download_loader
+
+GoogleDocsReader = download_loader('GmailReader')
```
Comment (jerryjliu) [inline loader_hub/gmail/README.md:28]: nit: Gmail (also if you want you could just remove the LlamaIndex and Langchain sections to make the readme more concise)

File: loader_hub/gmail/base.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    max_results: int = 10
+    service: Any
+
+    def load_data(
+        self
+    ) -> List[Document]:
+        """Load emails from the user's account
+
+        Args:
+            number_of_results (Optional[int]): the number of events to return. Defaults to 100.
```
Comment (jerryjliu) [inline loader_hub/gmail/base.py:28]: these aren't arguments in the load_data function, I would remove

**Review by jerryjliu — APPROVED** [review]
thanks! approving to unblock

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**